rero_ils/modules/libraries/api.py

```diff
@@ -58,3 +70,159 @@ def get_all_library_names(cls):
+                       day_only=False):
+        """Test the day has an exception."""
+        exception = open
+        for exception_date in exception_dates:
+            start_date = parser.parse(exception_date['start_date'])
+            repeat = exception_date.get('repeat')
+            if open:
+                # test for exceptios closed
+                if not exception_date['is_open']:
+                    has_period, is_in_period = self._is_in_period(
+                        date,
+                        exception_date
+                    )
+                    if has_period and is_in_period:
+                        exception = False
+                    if self._is_in_repeat(date, start_date, repeat):
+                        exception = False
+                    # we found a closing exception
+                    if not exception:
+                        return False
+            else:
+                # test for exceptions opened
+                if exception_date['is_open']:
+                    if self._is_in_repeat(date, start_date, repeat):
+                        exception = True
+                    if not exception and not day_only:
+                        exception = self._is_betweentimes(
+                            date.time(),
+                            exception_date.get('times', [])
+                        )
+                    return exception
+        return exception
+
+    def _has_is_open(self):
+        """Test if library has opening days."""
+        for opening_hour in self['opening_hours']:
+            if opening_hour['is_open']:
+                return True
+        exception_dates = self.get('exception_dates')
+        if exception_dates:
+            for exception_date in exception_dates:
+                if exception_date['is_open']:
+                    return True
+        return False
+
+    def is_open(self, date=datetime.now(), day_only=False):
+        """Test library is open."""
+        open = False
+        if isinstance(date, str):
+            date = parser.parse(date)
+        day_name = date.strftime("%A").lower()
+        for opening_hour in self['opening_hours']:
+            if day_name == opening_hour['day']:
+                open = opening_hour['is_open']
+                hours = opening_hour.get('times', [])
+                break
+        times_open = open
+        if open and not day_only:
+            times_open = self._is_betweentimes(date.time(), hours)
+        # test the exceptions
+        exception_dates = self.get('exception_dates')
+        if exception_dates:
+            exception = self._has_exception(
+                open=times_open,
+                date=date,
+                exception_dates=exception_dates,
+                day_only=day_only
+            )
+            if exception != times_open:
+                times_open = not times_open
+        return times_open
+
+    def next_open(self, date=datetime.now(), previous=False):
```

Comment:
not sure that default parameter calling function works... i.e. `datetime.now()`.

Reply:
I have tested it. It is working.